addons/api/addon/serializers/credential-library.js

```diff
@@ -36,23 +36,30 @@ export default class CredentialLibrarySerializer extends ApplicationSerializer {
     const value = super.serializeAttribute(...arguments);
     const { credential_type, type, credential_mapping_overrides, isNew } =
       snapshot.record;
+
     if (
       type === TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC &&
-      credential_mapping_overrides &&
-      credential_type &&
+      credential_mapping_overrides?.length &&
       !isNew
     ) {
       // API expects to send null to fields if it is undefined or deleted
-      json.credential_mapping_overrides = Object.keys(
-        options.mapping_overrides[credential_type],
-      ).reduce((obj, key) => {
-        if (credential_mapping_overrides[key]) {
-          obj[key] = credential_mapping_overrides[key];
-        } else {
-          obj[key] = null;
+      options.mapping_overrides[credential_type].forEach((item) => {
+        const exisiting = credential_mapping_overrides.find(
+          (obj) => obj.key === item,
+        );
+        if (!exisiting) {
+          credential_mapping_overrides.push({ key: item, value: null });
```

Comment:
The issue is with receiving the snapshot record data in array format (`credential_mapping_overrides`) before the `transformation` process. However, the JSON data must be in object format, so I need to do the transformation after setting the unused attributes to `null`. 

open to suggestions if there is a better approach to handle this

Reply:
Out of curiosity, does leaving it just as an array not work? I thought what would happen was it would go through this serializer and then the transformer on its way out, is that not how it works?

Reply:
It seems like the `serializeAttribute` method did not undergo transformation once we assigned it to the JSON. I moved this entire processing to the vault specific `serialize` method, which we already had. Since we are updating the serialized attribute, we didn't have to perform any additional transformation. Let me know your thoughts on that.

Reply:
Yeah that makes sense!